imap/caldav_alarm.c

```diff
@@ -809,6 +857,78 @@ static int has_alarms(void *data, struct mailbox *mailbox,
 
         syslog(LOG_DEBUG, "checking resource");
         for (; comp; comp = icalcomponent_get_next_component(ical, kind)) {
+            icalproperty *prop;
+
+            /* Disable alarms that fire too frequently */
+            for (prop =
+                     icalcomponent_get_first_property(comp, ICAL_RRULE_PROPERTY);
+                 prop;
+                 prop =
+                     icalcomponent_get_next_property(comp, ICAL_RRULE_PROPERTY)) {
+
+                struct icalrecurrencetype rrule = icalproperty_get_rrule(prop);
+                int recur_interval = rrule.interval;
+                char *bypart = "";
+                int disable = 0;
+
+                switch (rrule.freq) {
+                case ICAL_YEARLY_RECURRENCE:
+                case ICAL_MONTHLY_RECURRENCE:
+                case ICAL_WEEKLY_RECURRENCE:
+                    recur_interval *= 7;
```

Comment:
Maybe just always allow YEARLY, MONTHLY and WEEKLY without check? And state so in the documentation?

Reply:
I think we still want to check the BY parts for these frequencies.  FREQ=WEEKLY;BYSECOND=1,2,3,4,5 is ridiculous but possible